### ISSUE TITLE
ContainerInterfacePrivateServiceRule should report services that are implicitly private

### ISSUE THREAD

**Issue by nicwortel** [phpstan/phpstan-symfony#75]
[Since Symfony 3.4, services are private by default](https://symfony.com/blog/new-in-symfony-3-4-services-are-private-by-default). However, the [`XmlServiceMapFactory`](https://github.com/phpstan/phpstan-symfony/blob/master/src/Symfony/XmlServiceMapFactory.php) still assumes that a service is public if the `public` attribute hasn't been set:

https://github.com/phpstan/phpstan-symfony/blob/a6d87e0068741327f99cf3ec0e9c4f6809e602e2/src/Symfony/XmlServiceMapFactory.php#L48-L54

([the 3rd argument of the `Service` constructor is `bool $public`](https://github.com/phpstan/phpstan-symfony/blob/a6d87e0068741327f99cf3ec0e9c4f6809e602e2/src/Symfony/Service.php#L26)).

This means that the [`ContainerInterfacePrivateServiceRule`](https://github.com/phpstan/phpstan-symfony/blob/master/src/Rules/Symfony/ContainerInterfacePrivateServiceRule.php) will only report private services if they are explicitly marked as private in their service configuration, but not if they are implicitly private.

In order to correctly report private services being fetched from the container, this logic should be reversed so that a service without a `public` flag is considered private:

```php
$service = new Service(
	// ...
	isset($attrs->public) && (string) $attrs->public !== 'false',
	// ...
);
```

That will result in false positives for Symfony versions < 3.4, [but those versions are no longer supported by Symfony itself](https://symfony.com/releases). Alternatively, the `XmlServiceMapFactory` would have to determine the Symfony version and switch it's logic based on that, but I'm not sure if that's even possible...

**Comment by nicwortel** [comment]
@lookyman would you accept a PR that reverses the logic so services are considered private by default?

How would you like to handle old Symfony versions?

> That will result in false positives for Symfony versions < 3.4, but those versions are no longer supported by Symfony itself. Alternatively, the XmlServiceMapFactory would have to determine the Symfony version and switch it's logic based on that, but I'm not sure if that's even possible...

**Comment by lookyman** [comment]
I would, thanks. Also, we don't need to worry about older versions.

**Comment by jeroennoten** [comment]
I created #211 to resolve this.

**Comment by ondrejmirtes** [comment]
Fixed, merged, and released :)
